### DISCUSSION TITLE
City based routing

### DISCUSSION THREAD

**Discussion by vetalstar** [vercel/next.js#22680] [Ideas]
**Describe the feature you'd like to request**

For SEO requirements, you often need to have a city or county-based domain routing (or sub-path routing):
nyc.example.com
msk.example.com
ru.example.com

**Describe the solution you'd like**

I see the possibility of internationalized routing. It'll be very cool to have something like this but for user location.
Of course, we should be able to use our own endpoint to determine the location by IP.
It's also very important to save support of automatically statically optimized pages.

**Describe alternatives you've considered**

I don't see any alternatives right now but I don't have deep knowledge in Next.js

**Comment by leerob** [comment]
You can use `getServerSideProps` to determine what the sub-domain is. 

```
export async function getServerSideProps({ req, res }) {
  return {
    props: {
      host: req.headers.host
    }
  };
}
```

https://multi-domain-first.vercel.app/

**Reply by vetalstar** [reply]
@leerob As I understood I need to manually deploy all my domains?

**Reply by leerob** [reply]
There's a single application containing [n] number of domains. You wouldn't need to deploy them independently.

**Reply by vetalstar** [reply]
If so why can't I open https://multi-domain-third.vercel.app/ ?

**Reply by leerob** [reply]
Because I can't add a wildcard domain on vercel.app. You can, however, on your own domain (e.g. city.test.com, another-city.test.com).
